### DISCUSSION TITLE
After defining four pollutants "SS, COD, TN, TP", extracting pollutant concentrations failed.

### DISCUSSION THREAD

**Discussion by cui-hanwu** [pyswmm/pyswmm#552] [Q&A]
When I use this code from the example:

python

from pyswmm import Simulation, Nodes

with Simulation('tests/data/model_weir_setting.inp') as sim:
    j1 = Nodes(sim)["J1"]
    for step in sim:
        print(j1.depth)
I get an unknown error: "function 'swmm_getNodePollut' not found". How can I solve this?

**Comment by bemcdonnell** [comment]
@cui-hanwu Is there an additional method that you are calling that wasn't posted in this?  Calling for depth at J1 wouldn't throw an error for pollutants.  Can you 1. confirm the constituents are properly added to the INP file (check the report file and/or run it with EPA-SWMM's UI to test). 2. Can you share any addition code / model for your project?
